### PR TITLE
Support uploading large artifact for model deployment.

### PR DESCRIPTION
**Pull request by lu-ohai**
### Support uploading large artifact for model deployment.

- Added spec in `ModelDeploymentRuntime` class to support uploading large artifact
- Apply the API in `DataScienceModel` to upload artifact

### Example

```
ModelDeploymentContainerRuntime()
...
.with_model_uri("path_to_artifact")
.with_auth({"test_key":"test_value"})
.with_region("test_region")
.with_overwrite_existing_artifact(True)
.with_remove_existing_artifact(True)
.with_timeout(100)
.with_bucket_uri("test_bucket_uri")
...
```

### CODE REVIEW

File: ads/model/deployment/model_deployment.py [hunk 1]
```diff
@@ -1563,15 +1567,31 @@ def _build_model_deployment_configuration_details(self) -> Dict:
...
         if not model_id.startswith("ocid"):
-            model_id = OCIClientManager().prepare_artifact(
-                model_uri=runtime.model_uri,
-                properties=dict(
-                    display_name=self.display_name,
-                    compartment_id=self.infrastructure.compartment_id
-                    or COMPARTMENT_OCID,
-                    project_id=self.infrastructure.project_id or PROJECT_OCID,
-                ),
+            if ads_utils.folder_size(runtime.model_uri) > MAX_ARTIFACT_SIZE_IN_BYTES:
```
Comment (mrDzurb) [inline ads/model/deployment/model_deployment.py:1570]: Do we really need this validation? Wont this be captured by the DSC model by itself?

File: ads/model/deployment/model_deployment.py [hunk 2]
```diff
@@ -1563,15 +1567,31 @@ def _build_model_deployment_configuration_details(self) -> Dict:
...
+                if not runtime.bucket_uri:
+                    raise ValueError(
+                        f"The model artifacts size is greater than `{MAX_ARTIFACT_SIZE_IN_BYTES}`. "
+                        "The `bucket_uri` needs to be specified to copy artifacts to the object storage bucket. "
+                        "Example: `runtime.with_bucket_uri(oci://<bucket_name>@<namespace>/prefix/)`"
+                    )
+            
+            from ads.model.datascience_model import DataScienceModel
```
Comment (mrDzurb) [inline ads/model/deployment/model_deployment.py:1578]: Wouldn't it be better to move this import on top?

File: ads/model/deployment/model_deployment.py (hunk 1 above)
Comment (lu-ohai) [reply ads/model/deployment/model_deployment.py:1570]: Either way works, can remove this validation.

File: ads/model/deployment/model_deployment.py (hunk 2 above)
Comment (lu-ohai) [reply ads/model/deployment/model_deployment.py:1578]: I did this but seems this will cause circular import error

**Review by mrDzurb — APPROVED** [review]

**Review by qiuosier — APPROVED** [review]

**Pull request merged**